### PR TITLE
Add migration guide for 2.0.0

### PR DESCRIPTION
**Pull request by mohssenfathi**
Adding a migration guide to assist with updating from `0.1x.0` to 2.0.0-beta` (once released)

### CODE REVIEW

File: documentation/migration-guide-2.0.0.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# Uber iOS SDK 2.0.0 - Migration Guide
+The Uber Rides SDK has been revamped and renamed to the Uber iOS SDK. Among other things, this new release includes a simplified API, better security for mobile OAuth, and updated platform support.
+
+There are a few breaking changes that must be handled when updating the SDK. This guide will help you navigate the major code changes necessary to make the upgrade.
+
+## New + Updated Features
+### Swift Package Manager Support
+Swift Package Manager is now the preferred way to integrate the Uber iOS SDK. Support for other third party package managers has been dropped for this release.
```
Comment (alanpaivaa) [inline documentation/migration-guide-2.0.0.md:8]: What's the reasoning for this?

File: documentation/migration-guide-2.0.0.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+### Improved API
+Authenticating with Uber has now been moved behind a simplified API. Now, less setup and state management is required from the caller.
+
+### Auth Code Grant Flow + PKCE
+The Authorization Code Grant Flow is now the default method for requesting auth credentials. In addition, support for PKCE and client access token exchange has been added.
+
+Support for the Implicit Grand Flow has been removed.
```
Comment (alanpaivaa) [inline documentation/migration-guide-2.0.0.md:16]: possible typo: Grand -> Grant

File: documentation/migration-guide-2.0.0.md (hunk 1 above)
Comment (mohssenfathi) [reply documentation/migration-guide-2.0.0.md:8]: Mobile Foundations is strongly encouraging this change. If the impact is too large we may revisit this decision down the line, but for now their stance is that SPM should be the only option going forward.

File: documentation/migration-guide-2.0.0.md (hunk 2 above)
Comment (mohssenfathi) [reply documentation/migration-guide-2.0.0.md:16]: Thanks for catching!

**Review by alanpaivaa — APPROVED** [review]
LGTM

**Pull request merged**